### ISSUE TITLE
Require sources to be directory names in manifest targets

### ISSUE THREAD

**Issue by thomashoneyman** [purescript/registry-dev#316]
The `Manifest` targets key includes a list of dependencies for the target and a `sources` key that identifies what files should be included for compilation wrt that target:

https://github.com/purescript/registry/blob/335c07cb00f3dada632019996d9f66eed9c6079e/v1/Target.dhall#L18-L19

However, as [described in this comment](https://github.com/purescript/registry/pull/309#issuecomment-1022561203), there are a number of problems with just allowing globs of any kind to be included in the sources key. And so, as [proposed in this comment](https://github.com/purescript/registry/pull/309#issuecomment-1024797588) and [endorsed in this comment](https://github.com/purescript/registry/pull/309#issuecomment-1024923289), we should update the spec to assert that the `sources` key does _not_ contain a list of globs, but rather contains a list of directories where sources can be found.

In short, we'd specify that:

1. The `sources` key contains a list of directories (for example: `[ "src", "test" ]`).
2. Package managers are to understand that these directories imply that PureScript files can be found with the glob pattern `"directory/**/*.purs"`, such as `"src/**/*.purs"`, and this transformation should be made when calling the PureScript compiler.

As a followup:

1. Should we require directories to be top-level only, like `"src"`, or should we allow directory paths like `"src/lib"`?
2. We need to check how many legacy packages would be affected by this restriction, if any, by checking that sources only contain strings of the form `DIR/**/*.purs` and can be stripped to `DIR`.

**Comment by f-f** [comment]
> Should we require directories to be top-level only, like "src", or should we allow directory paths like "src/lib"?

How would the behaviour change between these two options?

**Comment by thomashoneyman** [comment]
It’s just whether you can specify only a top-level directory or whether you can specify a subdirectory.

**Comment by thomashoneyman** [comment]
In the second option we’d ignore everything in src except for the lib directory — but maybe we only accept a top level directory

**Comment by f-f** [comment]
I don't see why we shouldn't accept arbitrary sub-directories?

Though note that for the legacy packages we would only accept `"src"`, for compatibility with the current convention.
